config.go

```diff
@@ -0,0 +1,132 @@
+// Copyright (c) 2017 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"errors"
+	"io/ioutil"
+
+	"github.com/BurntSushi/toml"
+	log "github.com/Sirupsen/logrus"
+	vc "github.com/containers/virtcontainers"
+	"github.com/containers/virtcontainers/pkg/oci"
+)
+
+const (
+	defaultRuntimeLib           = "/var/lib/cc-runtime"
+	defaultRuntimeRun           = "/var/run/cc-runtime"
+	defaultruntimeConfiguration = "/etc/cc-runtime/configuration.toml"
+	defaultHypervisor           = vc.QemuHypervisor
+	defaultProxy                = vc.CCProxyType
+	defaultAgent                = vc.HyperstartAgent
+)
+
+const (
+	qemuLite = "qemu-lite"
+	qemu     = "qemu"
+)
+
+const (
+	hyperstart = "hyperstart"
+)
+
+var (
+	errUnknownHypervisor  = errors.New("unknown hypervisor")
+	errUnknownAgent       = errors.New("unknown agent")
+	errTooManyHypervisors = errors.New("too many hypervisor sections")
+	errTooManyAgents      = errors.New("too many agent sections")
+)
+
+type tomlConfig struct {
+	Hypervisor map[string]hypervisor
+	Agent      map[string]agent
+}
+
+type hypervisor struct {
+	Path   string
+	Kernel string
+	Image  string
+}
+
+type agent struct {
+	Ctl string
+	IO  string
+}
+
+func loadConfiguration(configPath string) (oci.RuntimeConfig, error) {
+	config := oci.RuntimeConfig{
+		HypervisorType: defaultHypervisor,
+		AgentType:      defaultAgent,
+		ProxyType:      defaultProxy,
+	}
+
+	if configPath == "" {
+		configPath = defaultruntimeConfiguration
+	}
+
+	configData, err := ioutil.ReadFile(configPath)
+	if err != nil {
+		return config, err
+	}
+
+	var tomlConf tomlConfig
+	_, err = toml.Decode(string(configData), &tomlConf)
+	if err != nil {
+		return config, err
+	}
+
+	log.Debugf("TOML configuration: %v", tomlConf)
+
+	if len(tomlConf.Hypervisor) != 1 {
+		return config, errTooManyHypervisors
+	}
+
+	if len(tomlConf.Agent) != 1 {
+		return config, errTooManyAgents
+	}
+
+	var hConfig vc.HypervisorConfig
+	var aConfig interface{}
+
+	for k, hypervisor := range tomlConf.Hypervisor {
+		switch k {
+		case qemu:
+			fallthrough
+		case qemuLite:
+			hConfig = vc.HypervisorConfig{
+				HypervisorPath: hypervisor.Path,
+				KernelPath:     hypervisor.Kernel,
+				ImagePath:      hypervisor.Image,
+			}
+			break
+		}
+	}
+
+	for k, agent := range tomlConf.Agent {
+		switch k {
+		case hyperstart:
+			aConfig = vc.HyperConfig{
+				SockCtlName: agent.Ctl,
+				SockTtyName: agent.IO,
```

Comment:
I would prefer there is no way for us to specify those sockets as they could be not unique. We already have a mechanism in agent implementation for Hyperstart which is creating some unique socket names based on the podID.